railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -47,10 +47,11 @@ def parse_type_and_options(type)
               return $1, limit: $2.to_i
             when /decimal\{(\d+)[,.-](\d+)\}/
               return :decimal, precision: $1.to_i, scale: $2.to_i
-            when /(references|belongs_to)\{(.+)\}/
+            when /(references|belongs_to)(\{(.+)\})?/
               type = $1
-              provided_options = $2.split(/[,.-]/)
+              provided_options = $3 ? $3.split(/[,.-]/) : {}
               options = Hash[provided_options.map { |opt| [opt.to_sym, true] }]
+              options[:required] = true
```

Comment:
I feel `parse` is a more suitable place to add this change.  

```diff 
diff --git a/railties/lib/rails/generators/generated_attribute.rb b/railties/lib/rails/generators/generated_attribute.rb
index 99c1bc4..3db183f 100644
--- a/railties/lib/rails/generators/generated_attribute.rb
+++ b/railties/lib/rails/generators/generated_attribute.rb
@@ -28,6 +28,7 @@ def parse(column_definition)
             if UNIQ_INDEX_OPTIONS.include?(has_index)
               attr_options[:index] = { unique: true }
             end
+            attr_options[:required] = true
           end 
```  

What do you think?